src/containerapp/azext_containerapp/custom.py

```diff
@@ -473,9 +473,9 @@ def create_containerapp(cmd,
     if not managed_env_info:
         raise ValidationError("The environment '{}' does not exist. Specify a valid environment".format(managed_env))
 
-    while not no_wait and "properties" in managed_env_info and managed_env_info["properties"]["provisioningState"].lower() in ["inprogress", "updating"]:
+    while not no_wait and safe_get(managed_env_info, "properties", "provisioningState").lower() in ["inprogress", "updating"]:
```

Comment:
if `safe_get(managed_env_info, "properties", "provisioningState")` return `None`, the `lower()` will throw error.

Recommend to set a default empty string:
`safe_get(managed_env_info, "properties", "provisioningState", default="")`